napari/layers/utils/color_manager.py

```diff
@@ -0,0 +1,643 @@
+    current_color : Optional[np.ndarray]
+        A (4,) color array for the color of the next items to be added.
+    mode : ColorMode
+        The mode for setting colors.
+
+        ColorMode.DIRECT: colors are set by passing color values to ColorManager.colors
+        ColorMode.COLORMAP: colors are set via the continuous_colormap applied to the
+                            color_properties
+        ColorMode.CYCLE: colors are set vie the categorical_colormap appied to the
+                         color_properties. This should be used for categorical
+                         properties only.
+     color_properties : Optional[ColorProperties]
+        The property values that are used for setting colors in ColorMode.COLORMAP
+        and ColorMode.CYCLE. The ColorProperties dataclass has 3 fields: name,
+        values, and current_value. name (str) is the name of the property being used.
+        values (np.ndarray) is an array containing the property values.
+        current_value contains the value for the next item to be added. color_properties
+        can be set as either a ColorProperties object or a dictionary where the keys are
+        the field values and the values are the field values (i.e., a dictionary that would
+        be valid in ColorProperties(**input_dictionary) ).
+    continuous_colormap : Colormap
+        The napari colormap object used in ColorMode.COLORMAP mode. This can also be set
+        using the name of a known colormap as a string.
+    contrast_limits : Tuple[float, float]
+        The min and max value for the colormap being applied to the color_properties
+        in ColorMonde.COLORMAP mode. Set as a tuple (min, max).
+    categorical_colormap : CategoricalColormap
+        The napari CategoricalColormap object used in ColorMode.CYCLE mode.
+        To set a direct mapping between color_property values and colors,
+        pass a dictionary where the keys are the property values and the
+        values are colors (either string names or (4,) color arrays).
+        To use a color cycle, pass a list or array of colors. You can also
+        pass the CategoricalColormap keyword arguments as a dictionary.
+    colors : np.ndarray
+        The colors in a Nx4 color array, where N is the number of colors.
+    """
+
+    # fields
+    current_color: Optional[np.ndarray] = None
+    mode: ColorMode = ColorMode.DIRECT
+    color_properties: Optional[ColorProperties] = None
+    continuous_colormap: Colormap = 'viridis'
+    contrast_limits: Optional[Tuple[float, float]] = None
+    categorical_colormap: CategoricalColormap = [0, 0, 0, 1]
+    colors: Array[float, (-1, 4)] = []
+
+    __equality_checks__ = {
+        'continuous_colormap': compare_colormap,
+        'categorical_colormap': compare_categorical_colormap,
+        'color_properties': compare_color_properties,
+        'contrast_limits': compare_contrast_limits,
+        'current_color': compare_colors,
+        'colors': compare_colors,
+    }
```

Comment:
Yeah, those are just placeholders until we solve #2200.  I agree it would be nice to put them somewhere else and not in `__equality_checks__`. For the fields that are `EventedModel` subclasses, I wonder if for now, it makes sense to add an `__eq__` method to the `EventedModel` subclass (e.g., `CategoricalColormap` and `Colormap`). It seems like that's code that could be reused by other `EventedModel`s that have them as fields and also keeps related code together.